src/main/java/org/codehaus/mojo/versions/DisplayDependencyUpdatesMojo.java

```diff
@@ -691,12 +691,19 @@ private DependencyManagement getProjectDependencyManagement( MavenProject projec
 
     private Optional<Segment> calculateUpdateScope()
     {
-        return !allowAnyUpdates
-                ? allowMajorUpdates ? of( MAJOR )
-                : allowMinorUpdates ? of( MINOR )
-                : allowIncrementalUpdates ? of( INCREMENTAL )
-                : empty()
-                : empty();
+        if ( !allowIncrementalUpdates && !allowMinorUpdates && !allowMajorUpdates && !allowAnyUpdates )
+        {
+            throw new IllegalArgumentException( "One of: allowAnyUpdates, allowMajorUpdates, allowMinorUpdates, "
+                    + "allowIncrementalUpdates must be true" );
+        }
+
+        return allowAnyUpdates && allowMajorUpdates && allowMinorUpdates
+                ? empty()
+                : allowMajorUpdates && allowMinorUpdates
+                    ? of( MAJOR )
+                    : allowMinorUpdates
+                        ? of( MINOR )
+                        : of( INCREMENTAL );
```

Comment:
difficult to reading ... maybe simple if/else 😄

Reply:
Okay, I prefer the Elvis operator, but I'll change it to the if ladder :)